tests/test_io/test_mat.py

```diff
@@ -80,7 +84,7 @@ def test_large_bounds(tmpdir: "py.path.local", model: "Model") -> None:
 
 @pytest.mark.skipif(scipy is None, reason="scipy unavailable")
 def test_read_rewrite_matlab_model(
-    tmpdir: "py.path.local", data_directory: str
+    compare_models: Callable, tmpdir: "py.path.local", data_directory: str
```

Comment:
Again, I'd modify the test to use
```suggestion
    compare_models: Callable, tmp_path: Path, data_directory: Path
```